plugins/services/src/js/components/forms/NetworkingFormSection.js

```diff
@@ -526,15 +526,28 @@ class NetworkingFormSection extends mixin(StoreMixin) {
   }
 
   getVirtualNetworks() {
+    const mesosContainer =
+      findNestedPropertyInObject(this.props.data, `container.type`) === "MESOS";
+    // Networks with subnet6 should be disabled when "UCR" container type is selected.
+    const virtualNetworkIsAvailable = (mesosContainer, subnet6) =>
+      !mesosContainer || !subnet6;
+
     return VirtualNetworksStore.getOverlays()
       .mapItems(overlay => {
         const name = overlay.getName();
 
         return {
+          enabled: overlay.info.enabled,
+          subnet6: overlay.getSubnet6(),
           text: `Virtual Network: ${name}`,
           value: `${CONTAINER}.${name}`
         };
       })
+      .filterItems(
+        virtualNetwork =>
+          virtualNetwork.enabled &&
+          virtualNetworkIsAvailable(mesosContainer, virtualNetwork.subnet6)
+      )
       .getItems()
       .map((virtualNetwork, index) => {
```

Comment:
🎨 Not related to this PR but there is no need to first map the data into an intermediate object to then map them to a component. Consider collapsing these steps to avoid unnecessary calls.